src/context.cc

```diff
@@ -197,16 +191,51 @@ void Context::change_buffer(Buffer& buffer)
 void Context::forget_buffer(Buffer& buffer)
 {
     m_jump_list.forget_buffer(buffer);
-    if (m_last_buffer.get() == &buffer)
-        m_last_buffer = nullptr;
 
     if (&this->buffer() != &buffer)
         return;
 
     if (is_editing() && has_input_handler())
         input_handler().reset_normal_mode();
 
-    change_buffer(m_last_buffer ? *m_last_buffer : BufferManager::instance().get_first_buffer());
+    auto last_buffer = this->last_buffer();
+    change_buffer(last_buffer ? *last_buffer : BufferManager::instance().get_first_buffer());
+}
+
+Buffer* Context::last_buffer() const
+{
+    // FIXME: convert JumpList to a vector and use find_if() with reverse iterators
```

Comment:
If you just expose a ConstArrayView from the jump list you can, you could even do `find_if(jump_list.list() | reverse(), <predicate>)` with the toolset exposed in ranges.hh